grails-app/controllers/org/pih/warehouse/inventory/StockTransferController.groovy

```diff
@@ -97,4 +98,34 @@ class StockTransferController {
         }
         redirect(action: "list")
     }
+
+    def deleteInboundOrOutboundReturn = {
+        Location currentLocation = Location.get(session.warehouse.id)
+        Order orderInstance = Order.get(params.orderId)
```

Comment:
`Order orderInstance = Order.get(params.orderId ?: params.id)`

Reply:
```
if (!orderInstance) {
    throw exception
}
```

Reply:
StockMovementDirection direction =  orderInstance?.getStockMovementDirection(currentLocation)

(and call `tockTransferService.deleteStockTransfer(params.orderId ?: params.id)` once and only make ifs for redirects)